src/components/EditableHeading/EditableHeading.tsx

```diff
@@ -0,0 +1,177 @@
+import { ComponentDefaultTestId, getTestId } from "../../tests/test-ids-utils";
+import React, { forwardRef, useEffect, useLayoutEffect, useMemo, useRef, useState } from "react";
+import VibeComponentProps from "../../types/VibeComponentProps";
+import Heading from "../Heading/Heading";
+import { withStaticProps } from "../../types";
+import styles from "./EditableHeading.module.scss";
+import { HeadingType, HeadingWeight } from "../Heading/HeadingConstants";
+import cx from "classnames";
+import { getStyle } from "../../helpers/typesciptCssModulesHelper";
+import { camelCase } from "lodash-es";
+import HiddenInputPlaceholder from "./HiddenInputPlaceholder";
+import { keyCodes } from "../../constants";
+import useMergeRefs from "../../hooks/useMergeRefs";
+import useClickableProps from "../../hooks/useClickableProps/useClickableProps";
+
+export interface EditableHeadingProps extends VibeComponentProps {
+  /** Heading text */
+  value: string;
+  /** Heading type */
+  type?: HeadingType;
+  /** Heading weight */
+  weight?: HeadingWeight;
+  /** onChange callback */
```

Comment:
I meant here to something that explains better what it does. for example in this case, `will be called whenever the current value changes to a non-empty value, `